src/actions/counter.js

```diff
@@ -1,36 +1,33 @@
-const CounterTypes = require('action-types/counter');
-
-exports.increment = (amount = 1) => {
-
-    return {
-        type: CounterTypes.COUNTER_INCREMENT,
-        payload: amount
-    };
-};
-
-
-/*  This is a thunk, meaning it is a function that immediately
-    returns a function for lazy evaluation. It is incredibly useful for
-    creating async actions, especially when combined with redux-thunk!
-
-    NOTE: This is solely for demonstration purposes. In a real application,
-    you'd probably want to dispatch an action of COUNTER_DOUBLE and let the
-    reducer take care of this logic.  */
-
-exports.doubleAsync = () => {
-
-    return (dispatch, getState) => {
-
-        return new Promise((resolve) => {
-
-            setTimeout(() => {
-
-                const count = getState().counter;
-                const double = exports.increment(count);
-
-                dispatch(double);
-                resolve();
-            }, 200);
-        });
-    };
-};
+const T = require('prop-types');
+const { makeActionCreator } = require('utils/redux-helpers');
+const { COUNTER: Types } = require('action-types');
+
+const actions = exports;
+
+// Notice a limitation here -- can't set a default of 1 to increment
+actions.increment = makeActionCreator(
```

Comment:
What benefits does converting this into a simple action type give?  Consistency in how the app is written, and the ability to tie into `after` functionality?  I assume it'd be a pain to fire off the original increment action because of the changes to types?

Just thinking about what it might be like to upgrade an app to this style.  Would we need to hit every action in one swoop, or is there a reasonable way to port it over time?

Reply:
> Why
Yeah just for consistency, to use both parts of these helpers -- `SIMPLE` and `ASYNC`

> Fire off the original increment action
Referring to the default setting of 1? If we want it there we can set it in initial state -- https://github.com/BigRoomStudios/strangeluv/pull/220/files#diff-ab974bcb2c9bcc1199dde0a37304b807R4

Reply:
> Just thinking about what it might be like to upgrade an app to this style. Would we need to hit every action in one swoop, or is there a reasonable way to port it over time?

Based upon my understanding, it could be done incrementally.  At the end of the day `actions.increment` is an action, either plain or a thunk.